test/yargs.js

```diff
@@ -2339,4 +2339,36 @@ describe('yargs dsl tests', () => {
         })
     })
   })
+
+  describe('onFinishCommand', () => {
+    it('use with promise', (done) => {
+      const result = 'noop-result'
+      yargs(['noop']).command('noop', 'a noop command', noop, () => {
+        return Promise.resolve(result)
+      }).onFinishCommand(async (commandResult) => {
+        commandResult.should.eql(result)
```

Comment:
If your command handler is long enough to execute (if you add a timer for example), this assertion should be ignored, as done() will have been called by parse callback before.

Reply:
👌